sycl/source/detail/context_impl.cpp

```diff
@@ -90,6 +90,9 @@ context_impl::~context_impl() {
       PI_CALL(piKernelRelease)(KernIt.second);
     PI_CALL(piProgramRelease)(ToBeDeleted);
   }
+  for (auto LibProg : MCachedLibPrograms) {
+    PI_CALL(piProgramRelease)(LibProg.second);
```

Comment:
Must be checked for non null value

Reply:
There should never be a null program, any incomplete program should be erased.
I'll add an assert here.